pkg/metrics/metrics.go

```diff
@@ -0,0 +1,275 @@
+			Measure:     CloudProviderOperationsErrorsCountM,
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{operationTypeKey},
+		},
+		&view.View{
+			Description: KubernetesAPIOperationsErrorsCountM.Description(),
+			Measure:     KubernetesAPIOperationsErrorsCountM,
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{operationTypeKey},
+		},
+	}
+
+	err := view.Register(views...)
+	return err
+}
+
+// record records the given measure
+func record(ctx context.Context, ms ...stats.Measurement) {
+	stats.Record(ctx, ms...)
+}
+
+// reporter is stats reporter in the context
+type reporter struct {
+	ctx context.Context
+}
+
+// Report records the given measure
+func (r *reporter) Report(ms ...stats.Measurement) {
+	record(r.ctx, ms...)
+}
+
+// NewReporter creates a reporter with new context
+func NewReporter() (*reporter, error) {
+	ctx, err := tag.New(
+		context.Background(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &reporter{ctx: ctx}, nil
+}
+
+// NewReporterAndReport creates a new instance of reporter and report given measurements
+func NewReporterAndReport(ms ...stats.Measurement) {
+
+	reporter, reporterError := NewReporter()
+
+	if reporterError != nil {
+		glog.Error(reporterError)
+	} else {
+		reporter.Report(ms...)
+	}
+}
+
+// ReportOperationStatusCount
+func (r *reporter) ReportOperationAndStatus(operationType string, statusCode string, ms ...stats.Measurement) error {
+
+	ctx, err := tag.New(
+		r.ctx,
+		tag.Insert(operationTypeKey, operationType),
+		tag.Insert(statusCodeKey, statusCode),
+	)
+	if err != nil {
+		return err
+	}
+	record(ctx, ms...)
+	return nil
+}
+
+// ReportMeasurementWithOperation records given measurement by operation type.
+func (r *reporter) ReportMeasurementWithOperation(operationType string, measurement stats.Measurement) error {
+
+	ctx, err := tag.New(
+		r.ctx,
+		tag.Insert(operationTypeKey, operationType),
+	)
+	if err != nil {
+		return err
+	}
+	record(ctx, measurement)
+	return nil
+}
+
+// RecordK8SAPIOperationError records the error in KubernetesAPIOperationsErrorsCountM
+func RecordK8SAPIOperationError(operation string) {
+
+	reporter, reporterError := NewReporter()
+	if reporterError != nil {
+		glog.Error(reporterError)
+	} else {
+		reporter.ReportMeasurementWithOperation(operation, KubernetesAPIOperationsErrorsCountM.M(1))
+	}
+}
+
+// RecordCloudProviderOperationError records the error in CloudProviderOperationsErrorsCountM
+func RecordCloudProviderOperationError(operation string) {
+
+	reporter, reporterError := NewReporter()
+	if reporterError != nil {
+		glog.Error(reporterError)
+	} else {
+		reporter.ReportMeasurementWithOperation(operation, CloudProviderOperationsErrorsCountM.M(1))
+	}
+}
+
+// RegisterAndExport register the views for the measures and exposeas  prometheus
+func RegisterAndExport(port string, log log.Logger) {
+
+	err := registerViews()
+
+	if err != nil {
+		log.Errorf("Failed to register views for metrics. error:%v", err)
```

Comment:
If there is an error registering view for metrics, we need to return the error.

Reply:
done.